src/Tester/StreamingTests/GeneratedStreamRecoveryTests.cs

```diff
@@ -0,0 +1,82 @@
+﻿    
+using System.Collections.Generic;
+using System.IO;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Orleans;
+using Orleans.Streams;
+using Orleans.TestingHost;
+using Tester.StreamingTests;
+using Tester.TestStreamProviders.Generator;
+using TestGrains;
+using UnitTests.Grains;
+using UnitTests.Tester;
+
+namespace UnitTests.StreamingTests
+{
+    [DeploymentItem("OrleansConfigurationForTesting.xml")]
+    [DeploymentItem("OrleansProviders.dll")]
+    [TestClass]
+    public class GeneratedImplicitSubscriptionStreamRecoveryTests : HostedTestClusterPerFixture
+    {
+        private static readonly string StreamProviderTypeName = typeof(GeneratorStreamProvider).FullName;
+        private const string StreamProviderName = GeneratedStreamTestConstants.StreamProviderName;
+
+        private readonly static GeneratorAdapterConfig AdapterConfig = new GeneratorAdapterConfig(StreamProviderName)
+        {
+            TotalQueueCount = 4,
+        };
+
+        private ImplicitSubscritionRecoverableStreamTestRunner runner;
+
+        public static TestingSiloHost CreateSiloHost()
+        {
+            return new TestingSiloHost(
+                new TestingSiloOptions
+                {
+                    StartFreshOrleans = true,
+                    SiloConfigFile = new FileInfo("OrleansConfigurationForTesting.xml"),
+                    AdjustConfig = config =>
+                    {
+                        var settings = new Dictionary<string, string>();
+                        // get initial settings from configs
+                        AdapterConfig.WriteProperties(settings);
+
+                        // add queue balancer setting
+                        settings.Add(PersistentStreamProviderConfig.QUEUE_BALANCER_TYPE, StreamQueueBalancerType.DynamicClusterConfigDeploymentBalancer.ToString());
+
+                        // add pub/sub settting
+                        settings.Add(PersistentStreamProviderConfig.STREAM_PUBSUB_TYPE, StreamPubSubType.ImplicitOnly.ToString());
+
+                        // register stream provider
+                        config.Globals.RegisterStreamProvider<GeneratorStreamProvider>(StreamProviderName, settings);
+
+                        // make sure all node configs exist, for dynamic cluster queue balancer
+                        config.GetConfigurationForNode("Primary");
```

Comment:
I'm not sure what to recommend here, as I'm not sure I understand what it's trying to accomplish. Can't we just set the configuration directing instead of having a base config and then updating it? or are these knobs not exposed in some other way?